### PR TITLE
Add basic support for jwt LDAP auth

### PR DESCRIPTION
**Pull request by nagem**
Allows users to authenticate with a JWT token distributed by the ldap-jwt web service. Support for multiple auth types will be added as an enhancement when a plan to continue supporting environment variable config can be created that works with an array of auth config options. 

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

### CODE REVIEW

**Comment by kofalt** [comment]
LGTM. Might want to have a gander at the LDAP library docs to make sure that `r.ok` is a sufficient check against all possible error returns.

**Comment by nagem** [comment]
@dpuccetti you can authenticate the same as always - just pass the jwt in the `Authorization` header instead of the Google token.

**Comment by ryansanford** [comment]
@nagem Can you add the new env variable to `sample.config` file?

**Comment by nagem** [comment]
@kofalt The source code of the tiny node service we're using is pretty simple, I don't see any way we'd get bit by that `r.ok` logic. Even if it did I doubt the return would have a valid email.

**Comment by nagem** [comment]
@ryansanford added.

**Comment by ryansanford** [comment]
@nagem Dev and Prod environment changes to default `SCITRAN_AUTH_AUTH_TYPE` correctly are complete.

Suggest setting default in `api/config.py` to align with the related existing defaults.

After that, LGTM.

File: api/config.py [hunk 1]
```diff
@@ -42,6 +42,7 @@
         'prefetch': False
     },
     'auth': {
+        'auth_type': 'google',
```
Comment (nagem) [inline api/config.py]: @ryansanford let me know if you meant something other than this in your latest comment.

**Pull request merged**